src/speculators/data_generation/preprocessing.py

```diff
@@ -196,6 +224,9 @@ def _preprocess_batch(
         log.warning(f"No conversations key found. Keys: {list(examples.keys())}")
         return results
 
+    # Check for mask support once per batch
+    supports_mask = _supports_assistant_mask(tokenizer)
+
```

Comment:
Read other comment first.

We don't need this check if we can allow `assistant_pattern` to be `None` and treat that as an indicator of whether the tokenizer accepts `return_assistant_tokens_mask=True` or not. This is also good because we probably don't want to be doing this check for every single batch.